types/queries.d.ts

```diff
@@ -111,24 +116,17 @@ export interface ByRoleOptions extends MatcherOptions {
   /**
    * Only considers elements with the specified accessible name.
    */
-  name?:
-    | RegExp
-    | string
-    | ((accessibleName: string, element: Element) => boolean)
+  name?: RegExp | string | MatcherFunction
```

Comment:
The original signature included useful type hints (e.g. `accessibleName`). These are lost with a generic `MatcherFunction`.

Could we instead use a concrete `NameMatcherFunction` to get type hints and keep the improved union?